pkg/apis/serving/v1beta1/v1beta1.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2020 kubeflow.org.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// NOTE: Boilerplate only.  Ignore this file.
+
+// Package v1beta1 contains API Schema definitions for the serving v1beta1 API group
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen=package,register
+// +k8s:conversion-gen=kfserving/pkg/apis/serving
+// +k8s:defaulter-gen=TypeMeta
+// +groupName=serving.kubeflow.org
+package v1beta1
```

Comment:
I was discussing this on another PR from Bloomberg, but I'll mention too you too.

Given how complex each of our specs are, I think we should be striving to separate these components into different packages to keep the code maintainable. However, they all need to be part of the same API Group. Everything should be part of `serving.kubeflow.org` and different resources will implement `v1alpha2` or `v1beta1`. The folder name "serving" used to corresp
ond to the group name, but we've evolved beyond that now.

Therefore, I think it probably makes sense to do something like this.
```
pkg/apis/v1alpha2.go // Defines alpha apis/schemes/GVK (group version kind)
pkg/apis/v1beta1.go // Defines beta apis/schemes/GVK
pkg/apis/inferencerouter/v1alpha1
pkg/apis/inferenceservice/v1alpha2
pkg/apis/inferenceservice/v1beta1
pkg/apis/trainedmodel/v1beta1
```
What do you think?

Reply:
sgtm,  as this is going to involve changes in many places can we do it in a follow up PR?

Reply:
Yeah definitely.